pkg/component/runtime/subscription.go

```diff
@@ -43,6 +43,16 @@ func newSubscriptionAll(ctx context.Context, manager *Manager) *SubscriptionAll
 	}
 }
 
+// NewSubscriptionAllWithChannel creates a SubscriptionAll using an existing channel
+// For Test purposes ONLY.
+func NewSubscriptionAllWithChannel(ctx context.Context, manager *Manager, evtChan chan ComponentComponentState) *SubscriptionAll {
```

Comment:
[Suggestion]
Is it really necessary to add a exported function for tests only?
Besides, from what I see, you could even get around it without this function, creating the `SubscriptionAll` directly on the test

Reply:
There is no way to create a SubscriptionAll object that returns a channel that can be controlled from the test code...

There are alternatives by mocking and changing the return types of a couple of interfaces but after talking to @michalpristas this seemed to be the preferred option